### DISCUSSION TITLE
Option to include code between chapters and appendices in Quarto formats

### DISCUSSION THREAD

**Discussion by danielvartan** [quarto-dev/quarto-cli#7454] [Q&A] [answered]
### Description

Changing the main template and partial templates in Quarto is great for customizing the output, but I believe they still need more options to provide a high degree of customization.

As far as I know, Quarto still doesn't allow different placement for chapters and appendices, bundling the two in a variable named `$body$`.

https://github.com/quarto-dev/quarto-cli/blob/1bfd06d1f59153ddd9785faadb1cf93467f91c57/src/resources/formats/pdf/pandoc/template.tex#L385

This can cause some problems. For example, if you're creating a book and want to programmatically build a Table of Contents (ToC) in a PDF between chapters and appendices using the heading feature of BibLaTeX `\printbibliography`, that can be very challenging.

```
\defbibheading{bay}$if(biblio-footnote)$[\bibnamewithfootnote]$else$[\bibname]$endif${
  \ifthenelse{\boolean{ABNTEXupperchapter}}{
    \setboolean{ABNTEXupperchapter}{false}
    \chapter*{#1}
    \markboth{#1}{#1}
    \setboolean{ABNTEXupperchapter}{true}
  }{
    \chapter*{#1}
    \markboth{#1}{#1}
  }

\printbibliography[heading=bay]
```

Of course, one can do something similar without making use of a greater level of customization by creating a "References" chapter before the appendices and using the `::: {#refs} ::::` chunk. However, that's not the goal here.

Another problem that arises from bundling together chapters and appendices is the difficulty in formatting the appendices section since Quarto also applies some formatting before rendering it.

```
# Chapters code end here

\cleardoublepage
\phantomsection
\addcontentsline{toc}{part}{Appendices}
\appendix

# Appendix code start here
```

If you're dealing with special LaTeX classes like [`abntex2`](https://github.com/abntex/abntex2), this can cause some trouble because it requires calling a new environment between the two for proper formatting.

These are just a few examples of why I believe the variable `$body$` should be unbundled.

**Comment by cscheid** [comment, marked as answer]
Unfortunately, `$body$` is a Pandoc feature, and we can't really change it.

**Reply by danielvartan** [reply]
I see. Thank you @cscheid !

**Comment by danielvartan** [comment]
One solution would be to create an `after-chapters.tex` partial.

A better solution would be to give the user control over where each part of the book will be placed.

**Comment by mcanouil** [comment]
Note that `$body` (from Pandoc) allows you to do whatever you want using LaTeX:
- https://github.com/quarto-ext/latex-environment
- raw blocks: `` ```{=latex} `` to add any LaTeX commands.

**Reply by danielvartan** [reply]
Thanks for the tips @mcanouil !
